{"uid":"b5a2e969f70004de","name":"test_full_cluster_restart[SIGKILL-/snap/charmed-postgresql/[0-9]*/usr/bin/patroni]","fullName":"tests.integration.ha_tests.test_self_healing#test_full_cluster_restart","historyId":"bc079a669521a660f5a86415d1c5b102","time":{"start":1712714641682,"stop":1712714844863,"duration":203181},"description":"This tests checks that a cluster recovers from a full cluster restart.\n\n    The test can be called a full cluster crash when the signal sent to the OS process\n    is SIGKILL.\n    ","descriptionHtml":"<p>This tests checks that a cluster recovers from a full cluster restart.</p>\n<pre><code>The test can be called a full cluster crash when the signal sent to the OS process\nis SIGKILL.\n</code></pre>\n","status":"broken","statusMessage":"tenacity.RetryError: RetryError[<Future at 0x7f4f932432b0 state=finished raised AssertionError>]","statusTrace":"Traceback (most recent call last):\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/tests/integration/ha_tests/helpers.py\", line 99, in are_writes_increasing\n    assert more_writes[member] > count, f\"{member}: writes not continuing to DB\"\nAssertionError: postgresql-0: writes not continuing to DB\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 340, in from_call\n    result: Optional[TResult] = func()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 240, in <lambda>\n    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 501, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 119, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 181, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 99, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 166, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/threadexception.py\", line 87, in pytest_runtest_call\n    yield from thread_exception_runtest_hook()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/threadexception.py\", line 63, in thread_exception_runtest_hook\n    yield\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 166, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/unraisableexception.py\", line 90, in pytest_runtest_call\n    yield from unraisable_exception_runtest_hook()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/unraisableexception.py\", line 65, in unraisable_exception_runtest_hook\n    yield\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 166, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/logging.py\", line 849, in pytest_runtest_call\n    yield from self._runtest_for(item, \"call\")\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/logging.py\", line 832, in _runtest_for\n    yield\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 166, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/capture.py\", line 883, in pytest_runtest_call\n    return (yield)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 166, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/skipping.py\", line 256, in pytest_runtest_call\n    return (yield)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 102, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 182, in pytest_runtest_call\n    raise e\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 172, in pytest_runtest_call\n    item.runtest()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1772, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 501, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 119, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 181, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 99, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 102, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 195, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 532, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/tests/integration/ha_tests/test_self_healing.py\", line 294, in test_full_cluster_restart\n    await are_writes_increasing(ops_test)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/tests/integration/ha_tests/helpers.py\", line 94, in are_writes_increasing\n    for attempt in Retrying(stop=stop_after_delay(60 * 3), wait=wait_fixed(3)):\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/tenacity/__init__.py\", line 347, in __iter__\n    do = self.iter(retry_state=retry_state)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/tenacity/__init__.py\", line 326, in iter\n    raise retry_exc from fut.exception()\ntenacity.RetryError: RetryError[<Future at 0x7f4f932432b0 state=finished raised AssertionError>]","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"ops_test","time":{"start":1712712736800,"stop":1712712736932,"duration":132},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"tmp_path_factory","time":{"start":1712712736800,"stop":1712712736800,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop","time":{"start":1712712736800,"stop":1712712736800,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"reset_restart_condition","time":{"start":1712714629797,"stop":1712714639603,"duration":9806},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"abort_on_fail","time":{"start":1712714629797,"stop":1712714629797,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"continuous_writes","time":{"start":1712714629797,"stop":1712714629797,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"loop_wait","time":{"start":1712714639603,"stop":1712714641682,"duration":2079},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"afterStages":[{"name":"loop_wait::finalizer","time":{"start":1712714844901,"stop":1712714844935,"duration":34},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"reset_restart_condition::finalizer","time":{"start":1712714844936,"stop":1712714852850,"duration":7914},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"continuous_writes::finalizer","time":{"start":1712714852850,"stop":1712715156675,"duration":303825},"status":"failed","statusMessage":"AssertionError: Unable to clear up continuous_writes table\nassert 'False' == 'True'\n  \n  - True\n  + False\n","statusTrace":"  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/allure_commons/_allure.py\", line 221, in __call__\n    return self._fixture_function(*args, **kwargs)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 302, in finalizer\n    event_loop.run_until_complete(async_finalizer())\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 294, in async_finalizer\n    await gen_obj.__anext__()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/tests/integration/ha_tests/conftest.py\", line 29, in continuous_writes\n    for attempt in Retrying(stop=stop_after_delay(60 * 5), wait=wait_fixed(3), reraise=True):\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/tenacity/__init__.py\", line 347, in __iter__\n    do = self.iter(retry_state=retry_state)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/tenacity/__init__.py\", line 325, in iter\n    raise retry_exc.reraise()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/tenacity/__init__.py\", line 158, in reraise\n    raise self.last_attempt.result()\n  File \"/usr/lib/python3.10/concurrent/futures/_base.py\", line 451, in result\n    return self.__get_result()\n  File \"/usr/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n    raise self._exception\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/tests/integration/ha_tests/conftest.py\", line 37, in continuous_writes\n    assert action.results[\"result\"] == \"True\", \"Unable to clear up continuous_writes table\"\n","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":true,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"abort_on_fail::0","time":{"start":1712715156675,"stop":1712715156676,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test::finalizer","time":{"start":1712716996190,"stop":1712716997103,"duration":913},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::2","time":{"start":1712716997105,"stop":1712716997105,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_close_event_loop","time":{"start":1712716997105,"stop":1712716997105,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_provide_clean_event_loop","time":{"start":1712716997105,"stop":1712716997107,"duration":2},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"group(1)"},{"name":"tag","value":"asyncio"},{"name":"parentSuite","value":"tests.integration.ha_tests"},{"name":"suite","value":"test_self_healing"},{"name":"host","value":"fv-az887-658"},{"name":"thread","value":"8477-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.integration.ha_tests.test_self_healing"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"process","value":"'/snap/charmed-postgresql/[0-9]*/usr/bin/patroni'"},{"name":"signal","value":"'SIGKILL'"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":2,"broken":4,"skipped":0,"passed":51,"unknown":0,"total":57},"items":[{"uid":"72584ceca0efed1","reportUrl":"../2416//#testresult/72584ceca0efed1","status":"passed","time":{"start":1712628121330,"stop":1712628171831,"duration":50501}},{"uid":"a8796fabb4df8887","reportUrl":"../2409//#testresult/a8796fabb4df8887","status":"passed","time":{"start":1712541448240,"stop":1712541515847,"duration":67607}},{"uid":"4ef3226e6ef26c40","reportUrl":"../2408//#testresult/4ef3226e6ef26c40","status":"passed","time":{"start":1712455600578,"stop":1712455658440,"duration":57862}},{"uid":"c0ddac55a1fba10e","reportUrl":"../2407//#testresult/c0ddac55a1fba10e","status":"passed","time":{"start":1712370493662,"stop":1712370583669,"duration":90007}},{"uid":"ef574196930919e4","reportUrl":"../2362//#testresult/ef574196930919e4","status":"passed","time":{"start":1712282412943,"stop":1712282477096,"duration":64153}},{"uid":"98e16a85230778de","reportUrl":"../2343//#testresult/98e16a85230778de","status":"passed","time":{"start":1712195934819,"stop":1712195988519,"duration":53700}},{"uid":"71e5b577ba846bb7","reportUrl":"../2330//#testresult/71e5b577ba846bb7","status":"passed","time":{"start":1712109748379,"stop":1712109793134,"duration":44755}},{"uid":"2062272398407a6e","reportUrl":"../2327//#testresult/2062272398407a6e","status":"passed","time":{"start":1712023238594,"stop":1712023276410,"duration":37816}},{"uid":"fbf9d0408f42a28b","reportUrl":"../2324//#testresult/fbf9d0408f42a28b","status":"passed","time":{"start":1711937043881,"stop":1711937214154,"duration":170273}},{"uid":"a1fd7f60ade48279","reportUrl":"../2323//#testresult/a1fd7f60ade48279","status":"passed","time":{"start":1711850434115,"stop":1711850470899,"duration":36784}},{"uid":"4adb39f571373aca","reportUrl":"../2322//#testresult/4adb39f571373aca","status":"passed","time":{"start":1711765537557,"stop":1711765629239,"duration":91682}},{"uid":"78f4144867c974eb","reportUrl":"../2318//#testresult/78f4144867c974eb","status":"passed","time":{"start":1711677564558,"stop":1711677602724,"duration":38166}},{"uid":"83362a32b8a8390c","reportUrl":"../2315//#testresult/83362a32b8a8390c","status":"passed","time":{"start":1711591291173,"stop":1711591330174,"duration":39001}},{"uid":"6cd6311121d0c136","reportUrl":"../2313//#testresult/6cd6311121d0c136","status":"passed","time":{"start":1711504793834,"stop":1711504828430,"duration":34596}},{"uid":"98caa2cdb5482aa0","reportUrl":"../2309//#testresult/98caa2cdb5482aa0","status":"passed","time":{"start":1711418415891,"stop":1711418450896,"duration":35005}},{"uid":"d19cd8a1b97b0071","reportUrl":"../2306//#testresult/d19cd8a1b97b0071","status":"passed","time":{"start":1711331947548,"stop":1711331999029,"duration":51481}},{"uid":"a7ab8fde4ed7089","reportUrl":"../2305//#testresult/a7ab8fde4ed7089","status":"passed","time":{"start":1711245386232,"stop":1711245423597,"duration":37365}},{"uid":"c862df4676e57af4","reportUrl":"../2302//#testresult/c862df4676e57af4","status":"passed","time":{"start":1711159399945,"stop":1711159439813,"duration":39868}},{"uid":"4b8f9a58b7ca25e4","reportUrl":"../2278//#testresult/4b8f9a58b7ca25e4","status":"passed","time":{"start":1711074368691,"stop":1711074403943,"duration":35252}},{"uid":"53180b886ea45e5f","reportUrl":"../2273//#testresult/53180b886ea45e5f","status":"failed","statusDetails":"AssertionError: members report different lists of cluster members.","time":{"start":1710986452881,"stop":1710986499704,"duration":46823}}]},"tags":["asyncio","group(1)"]},"source":"b5a2e969f70004de.json","parameterValues":["'/snap/charmed-postgresql/[0-9]*/usr/bin/patroni'","'SIGKILL'"]}